src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Lookup.cs

```diff
@@ -47,6 +48,23 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
             // The return type is dictated by the last argument (projection) if one exists. Otherwise it's based on first argument (source).
             returnType = args.Length == 2 ? argTypes[0].ToRecord() : argTypes[2];
 
+            // Ensure that the arg at index 1 is boolean or can coerece to boolean if they are OptionSetValues.
+            if (DType.Boolean.Accepts(argTypes[1]))
```

Comment:
Empty code blocks are a little sketchy. Either you could `return fvalid` here, or just remove it, and change line 62 to `else if (!Dtype.Boolean.Accepts(ArgTypes[1])